src/app/content/highlights/hooks/createHighlight.ts

```diff
@@ -16,10 +14,13 @@ export const hookBody: ActionHookBody<typeof createHighlight> =
     try {
       await highlightClient.addHighlight({highlight: payload});
     } catch (error) {
-      Sentry.captureException(error);
-
-      dispatch(addToast(toastMessageKeys.higlights.failure.create, {destination}));
       dispatch(receiveDeleteHighlight(payload as unknown as Highlight, {...meta, revertingAfterFailure: true}));
+
+      if (error instanceof CustomApplicationError) {
```

Comment:
I've marked `UnauthenticatedError` and `BookNotFoundError` as `CustomApplicationError`
If we would add any new errors to the API clients then it would be handled by this if statement